### ISSUE TITLE
Embedding

### ISSUE THREAD

**Issue by cirdeCyL** [JiangBioLab/DLpTCR#2]
Good day,

I was wondering in the paper you tried out different embedding in combination with your submodels.
But in code you just have 2 files for the embedding:

1. Train_Test_Onehot_Chem_Feature_Extraction.py
2. Train_Test_PCA_Feature_Extraction.py

By reading the code in line 104 of Train_Test_Onehot_Chem_Feature_Extraction.py you concat the onehot with the chem. prop vectors. 
Is this the "Fusion feature encoding combined with one-hot and PCPs" (see Method part of your article)?
This embedding performed the worst of all the embedding (Figure 2 of your article).
Is the onehot embedding missing in this repository, because it was too trivial to share, or did I miss it somewhere?

 Thank you a lot.

Cheers,
Cedric

**Comment by JiangBioLab** [comment]
Thanks for your interest in DLpTCR!
To answer your first question, the data processing step in line 104 of Train_Test_Onehot_Chem_Feature_Extraction.py is indeed the " Fusion feature encoding combined with one-hot and PCPs" proposed in the article. I also checked the code immediately. Since the column names in the CSV file are uniformly modified when the code is finally submitted, some content needs to be modified in the Train_Test_Onehot_Chem_Feature_Extraction.py so that the CSV file can be read. Please see the changed file.
Regarding your second question, the onehot embedding is not missing. Instead, after reading the matrix file of the fusion feature encoding combined with one-hot and PCPs, only the onehot encoded content is selected as the input of the model, you can view the line 51 of the file /code/fold/CNN_A_fold_onehot.py.

---
42 def CNN_onehot(Dropout1=0,Epochs= 20,Batch_size=64):
...
51     X = Feature_test[:,0:29,:]
---

**Comment by cirdeCyL** [comment]
AH perfect. I see.
What I was missing a bit in the paper is that you have tried out many different hyperparameters. I think in the paper you talked about 50 different setup of hyperparameters, but I didn't find those hyperparameters anywhere on the paper or in the supplementary.
Do you recommend me to use the default values or did I just missed them.
Cheers, Cedric

**Comment by cirdeCyL** [comment]
Hey as a Reminder, I still would like to ask for the hyperparameter used.

**Comment by JiangBioLab** [comment]
> Hey as a Reminder, I still would like to ask for the hyperparameter used.

I'm really sorry that I missed your last email.
It may be that the description in our article is not clear and let you have the misunderstanding.We train the model 50 times using the final set of hyperparameters,not 50 different setup of  hyperparameters.Because we want to know the prediction performance of the model under this final set of hyperparameter combinations without setting the random seed, so as to give some reference for researchers who repeat our experiments (according to my tests, under the same combination of hyperparameters, even if the random seed is fixed, using GPU training does not ensure consistent prediction results for each model).
So you can use the default hyperarguments in our code.
Best wish for you!  
: )

**Comment by cirdeCyL** [comment]
Perfect answer. This cleared for me the issue.
Thank you for still replying.
Cheers
